### ISSUE TITLE
CI: code coverage summary

### ISSUE THREAD

**Issue by ozgurakgun** [conjure-cp/conjure-oxide#47]
@niklasdewally 

I was just looking at @Kieranoski702's PR and noticed this. I understand the "generate code coverage" action cannot deply the report - and we don't want it to for PRs - but can it print some stats in the info bit like the doc coverage ones do?

See this run as an example https://github.com/conjure-cp/conjure-oxide/actions/runs/6763169021

**Comment by ozgurakgun** [comment]
(and this for the doc coverage summary in case anyone doesn't know what I am talking about: https://github.com/conjure-cp/conjure-oxide/actions/runs/6763169024)

**Comment by niklasdewally** [comment]
I think this is a simple fix. There's a step in the coverage report deploy workflow that creates the summary table - this can be copied to the generate script too

**Comment by niklasdewally** [comment]
It seems to be broken on PRs generally - maybe we can fix later when we meet?

**Comment by niklasdewally** [comment]
For context, the action currently should deploy coverage for PRs in a secure way, but if you don't want this we can turn it off.

**Comment by ChrisJefferson** [comment]
In terms of security, just in case you didn't know, actions should not run for first time commuters until they are approved ( https://github.blog/2021-04-22-github-actions-update-helping-maintainers-combat-bad-actors/ ). Therefore you don't need to worry about a "drive by" person submitting an evil PR (they could of course do it subtly, so we don't notice and approve actions).

**Comment by ozgurakgun** [comment]
Thanks Chris. I noticed Github asking for approvals like this, but never looked into it.

I don't mind PRs uploading coverage reports, even better if we can fix that. That is as long as they don't edit the latest one (which should maybe called main after the name of the branch to avoid confusion?)

**Comment by niklasdewally** [comment]
Seems like deploy was probably fine all along,issue might be with how i got sha for PRs?
https://stackoverflow.com/a/68068674

TLDR; 

GITHUB_SHA points to the eventual PR merge commit, not to the head sha of the PR, or something like that...

**Comment by niklasdewally** [comment]
https://www.kenmuse.com/blog/the-many-shas-of-a-github-pull-request/
😢

**Comment by niklasdewally** [comment]
@ozgurakgun I think this can be closed
